### PR TITLE
Enabled partial checkpoint loading

### PR DESCRIPTION
**Pull request by danieltudosiu**
### Description
Allowing partial loading of a model via strict=False.

fixes https://github.com/Project-MONAI/MONAI/issues/1891

### Status
Ready

### Types of changes
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by danieltudosiu** [comment]
@Nic-Ma if I check the error of the check and I do what it asks I get

```
AttributeError: module 'ignite' has no attribute '__version__'
```

I installed the requirements based on requirements-dev.txt.

What should I do?

**Comment by danieltudosiu** [comment]
@wyli  I am using black locally as well. I always format my files with black before pushing them :-?

**Comment by wyli** [comment]
> @Nic-Ma if I check the error of the check and I do what it asks I get
> 
> ```
> AttributeError: module 'ignite' has no attribute '__version__'
> ```
> 
> I installed the requirements based on requirements-dev.txt.
> 
> What should I do?

could you show the full traceback? I can look into it

**Comment by danieltudosiu** [comment]
I checked and I have pytorch-ignite==0.4.4

Here it is:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/danieltudosiu/projects/MONAI/monai/__init__.py", line 43, in <module>
    load_submodules(sys.modules[__name__], False, exclude_pattern=excludes)
  File "/home/danieltudosiu/projects/MONAI/monai/utils/module.py", line 75, in load_submodules
    mod = import_module(name)
  File "/usr/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/danieltudosiu/projects/MONAI/monai/engines/__init__.py", line 12, in <module>
    from .evaluator import EnsembleEvaluator, Evaluator, SupervisedEvaluator
  File "/home/danieltudosiu/projects/MONAI/monai/engines/evaluator.py", line 17, in <module>
    from monai.engines.utils import IterationEvents, default_prepare_batch
  File "/home/danieltudosiu/projects/MONAI/monai/engines/utils.py", line 22, in <module>
    EventEnum, _ = optional_import("ignite.engine", "0.4.4", exact_version, "EventEnum")
  File "/home/danieltudosiu/projects/MONAI/monai/utils/module.py", line 201, in optional_import
    if not version_args and version_checker(pkg, f"{version}"):
  File "/home/danieltudosiu/projects/MONAI/monai/utils/module.py", line 109, in exact_version
    return bool(the_module.__version__ == version_str)
AttributeError: module 'ignite' has no attribute '__version__'
```

**Comment by wyli** [comment]
perhaps try `pip uninstall ignite` :)

**Comment by danieltudosiu** [comment]
Why? What's the logics? Isn't pytorch-ignite an optional dependency of MONAI?

If I run "pip list | grep ignite" I get:
```
pytorch-ignite                0.4.4
```

Isn't this supposed to be what's required for us to develop by using Ignite infrastructure?

**Comment by wyli** [comment]
thanks @danieltudosiu 

> Why? What's the logics? Isn't pytorch-ignite an optional dependency of MONAI?
> 
> If I run "pip list | grep ignite" I get:
> 
> ```
> pytorch-ignite                0.4.4
> ```
> 
> Isn't this supposed to be what's required for us to develop by using Ignite infrastructure?

there's another package named ignite https://pypi.org/project/ignite the folders can be messed up...

**Comment by danieltudosiu** [comment]
What else can it be? Since the checks are passing it means it is a local issue, not a MONAI one.

File: monai/handlers/checkpoint_loader.py [hunk 1]
```diff
@@ -82,7 +86,7 @@ def __call__(self, engine: Engine) -> None:
 
         # save current max epochs setting in the engine, don't overwrite it if larger than max_epochs in checkpoint
         prior_max_epochs = engine.state.max_epochs
-        Checkpoint.load_objects(to_load=self.load_dict, checkpoint=checkpoint)
+        Checkpoint.load_objects(to_load=self.load_dict, checkpoint=checkpoint, **{"strict": self.strict})
```
Comment (Nic-Ma) [inline monai/handlers/checkpoint_loader.py:89]: I would suggest to simplify this line to:
```py
Checkpoint.load_objects(to_load=self.load_dict, checkpoint=checkpoint, strict=self.strict)
```
Others look good to me.

Thanks.

**Review by Nic-Ma — APPROVED** [review]

**Comment by vfdev-5** [comment]
> What else can it be? Since the checks are passing it means it is a local issue, not a MONAI one.

@danieltudosiu it could be explaned by https://github.com/Project-MONAI/MONAI/pull/1930 where `get_package_version` before this PR removed the package from sys modules and thus there were a similar error while importing modules.

**Comment by danieltudosiu** [comment]
@vfdev-5  Thanks for pointing it to me. Will pull the latest commit from MONAI.

**Pull request merged**